core/auth.py

```
@@ -0,0 +1,685 @@
+    def _hash_api_key(self, api_key: str) -> str:
+        """Hash API key for storage."""
+        return hashlib.sha256(api_key.encode()).hexdigest()
+
+    def _verify_api_key(self, provided_key: str, stored_hash: str) -> bool:
+        """Verify API key against stored hash."""
+        return secrets.compare_digest(self._hash_api_key(provided_key), stored_hash)
+
+    def create_user(self, username: str, email: str, role: UserRole, password: Optional[str] = None) -> User:
+        """Create a new user account."""
+        if any(u.username == username for u in self._users.values()):
+            raise ValueError(f"User {username} already exists")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)